meshroom/ui/qml/Viewer/Viewer2D.qml

```diff
@@ -219,34 +219,47 @@ FocusScope {
             return sourceExternal;
         }
         if (_reconstruction && (!displayedNode || outputAttribute.name == "gallery")) {
-            return getViewpointPath(_reconstruction.selectedViewId);
+            return Filepath.stringToUrl(getViewpointAttribute("path",_reconstruction.selectedViewId));
         }
-        return getFileAttributePath(displayedNode, outputAttribute.name, _reconstruction ? _reconstruction.selectedViewId : -1);
+
+        var viewId = -1;
+        var intrinsicId = -1;
+        var fileName = -1;
+        if (_reconstruction) {
+            viewId = _reconstruction.selectedViewId;
+            intrinsicId = getViewpointAttribute("intrinsicId", viewId);
+            fileName = Filepath.removeExtension(Filepath.basename(Filepath.stringToUrl(getViewpointAttribute("path",viewId))));
+        }
+        var patterns = {"<VIEW_ID>": viewId,"<INTRINSIC_ID>": intrinsicId, "<FILENAME>": fileName}
+        return getFileAttributePath(displayedNode, outputAttribute.name,patterns);
     }
 
-    function getFileAttributePath(node, attrName, viewId) {
+    function getFileAttributePath(node, attrName, patterns) {
         // get output attribute with matching name
-        // and parse its value to get the image filepath
+        // and parse with the patterns present in the patterns dict
         if (!node)
             return "";
         for (var i = 0; i < node.attributes.count; i++) {
             var attr = node.attributes.at(i);
             if (attr.name == attrName) {
-                let pattern = String(attr.value).replace("<VIEW_ID>", viewId);
-                let path = Filepath.globFirst(pattern);
+                let path = String(attr.value)
+                for (var key in patterns) {
+                    if (patterns.hasOwnProperty(key)) {
```

Comment:
I think I'm missing the point, why do we need `hasOwnProperty` here ? 😅

Reply:
Just took the snipet from [stackoverflow](https://stackoverflow.com/questions/684672/how-do-i-loop-through-or-enumerate-a-javascript-object) saying that one should verify the key is a property of the object 
I can remove this if you want to !

Reply:
Thanks for the reference, that's fine by me let's keep it !

Reply:
javascript hell